### PR TITLE
Wait for control plane node to be ready after joining the cluster

### PR DESCRIPTION
**Pull request by aojea**
It can happen that the control plane node is not completely ready after joining the cluster,
If one worker node tries to join against a control plane node that's not ready, it fails to join thus the cluster creation fails.
This is a workaround to wait until the control node is ready after it joins the cluster before joining new nodes.

Fixes: https://github.com/kubernetes-sigs/kind/issues/588

### CODE REVIEW

File: pkg/cluster/internal/create/actions/kubeadmjoin/join.go [hunk 1]
```diff
@@ -211,6 +212,16 @@ func runKubeadmJoinControlPlane(
 		return errors.Wrap(err, "failed to join a control plane node with kubeadm")
 	}
 
+	// Wait for the node to be Ready
+	// TODO: remove once https://github.com/kubernetes-sigs/kind/issues/588 is fixed
+	// kubeadm join should guarantee that the cluster is ready
```
Comment (aojea) [inline pkg/cluster/internal/create/actions/kubeadmjoin/join.go]: @fabriziopandini @neolit123 I don't know if this is true 😅 , should `kubeadm join`, guarantee that?

File: pkg/cluster/internal/create/actions/kubeadmjoin/join.go [hunk 2]
```diff
@@ -211,6 +212,16 @@ func runKubeadmJoinControlPlane(
...
+	startTime := time.Now()
+	// TODO: currently hardcoded to 10s
+	isReady := nodes.WaitForReady(node, startTime.Add(10*time.Second))
```
Comment (thomastaylor312) [inline pkg/cluster/internal/create/actions/kubeadmjoin/join.go]: Super new to the code so I could be completely off, but isn't there a [`WaitForReady`](https://godoc.org/sigs.k8s.io/kind/pkg/cluster/create#WaitForReady)  option already that could be used here?

Comment (aojea) [reply pkg/cluster/internal/create/actions/kubeadmjoin/join.go]: yeah, indeed this function `nodes.WaitForReady` is used by the one you mentioned.
In this case we are looking for a specific node so I think that's better to use this one,

File: pkg/cluster/internal/create/actions/kubeadmjoin/join.go (hunk 1 above)
Comment (fabriziopandini) [reply pkg/cluster/internal/create/actions/kubeadmjoin/join.go]: @aojea see my comments on the issue https://github.com/kubernetes-sigs/kind/issues/588#issuecomment-499939113

In a nutshell Kubeadm is not responsible, but it is the kubelet. Additional, it seems also that the API server does not detected properly when the etcd instance is ready

Comment (neolit123) [reply pkg/cluster/internal/create/actions/kubeadmjoin/join.go]: > kubeadm join should guarantee that the cluster is ready

well kubeadm can look at pod and endpoint state, but the cluster as a whole - a bit tricky.
kinder currently waits for these pods + the node ready status, as @fabriziopandini  mentioned:
https://github.com/kubernetes/kubeadm/blob/62556834c87e34004ac84c17b2f2c68b5c4f3b22/kinder/pkg/actions/waiter.go#L32-L44

given HA join consistently does not fail using kind 0.2.0 as seen here:
https://k8s-testgrid.appspot.com/sig-cluster-lifecycle-kubeadm#kubeadm-kind-master
i'm trying to get to the bottom of the problem instead - i.e. finding a change in kind that helped the problem surface.

Comment (aojea) [reply pkg/cluster/internal/create/actions/kubeadmjoin/join.go]: kind is much faster with the latest versions, the fact that adding delays solve the problem or at least reduce them makes me think that´s tightly related to that

**Comment by BenTheElder** [comment]
Code looks fine FWIW but I'm not convinced we've fully root-caused the issue we're attempting to address yet, and I'm not a fan of adding arbitrary busy waits.

**Comment by aojea** [comment]
Definitely not the right approach

**Pull request closed without merging**